linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/ansible/shared.yml

```diff
@@ -9,7 +9,8 @@
     name: "{{ item }}"
     state: present
   with_items:
-    - NetworkManager
+    #- NetworkManager
+    - "{{{ platform_package_overrides.NetworkManager }}}"
```

Comment:
@marcusburghardt I don't think we normally do this, should we?

Also this ansible does not follow the ubuntu remediation or follow the advice from benchmarks on how to do it, either you should try to create a ubuntu specific ansible or add the missing bits here.

Reply:
I missed the old part that I commented out, if that's what you're referring to. 
@dodys I'm not sure I understand what you're referring to here though:

>Also this ansible does not follow the ubuntu remediation or follow the advice from benchmarks on how to do it, either you should try to create a ubuntu specific ansible or add the missing bits here.

Can you clairfy?

Reply:
this ansible remediation only talks about network-manager, it does not cover cases where you don't have it, see the bash remediation.

Reply:
Are you saying this pull-request will never be approved because it doesn't match the BASH remediation?

Reply:
No, I'm requesting changes to this PR so it can be approved. 
See bash:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/bash/ubuntu.sh

Reply:
Wouldn't that be more of an additional feature or enhancement? My pull-request fixes a failing play. I didn't add the ansible portion that doesn't match but fixed the existing one that fails because of the package name. It seems like creating parity between the BASH and ansible remediations is a bit more involved doesn't it? Maybe that should be a separate issue?

Reply:
No, the way it is right now it doesn't align to the expectation of the certification.
Your PR is installing network manager for systems that don't have network manager, therefore it needs to be correctly implemented.

Reply:
I don't disagree with you about the correct implementation, however prior to my pull-request, was this correctly implemented?

I do disagree that my PR installs network manager. The play prior to me touching it would have done this if it hadn't failed on the name difference between NetworkManager and network-manager. Without my PR, all plays after this never run because the package NetworkManager doesn't exist.

I'm not against bringing the ansible play into compliance with the BASH implementation, but the criteria for the acceptance of this PR seems a bit like not fixing a leaking sink because your house needs a new roof.

Reply:
no, it was not, but it also does not hurt the users with systems without network-manager. What you are suggesting here is actually introducing a regression, instead of the correct solution.
If a sink is leaking, you won't fix it by breaking someone's else sink.

Reply:
@marcusburghardt Do you have any thoughts here since it seems we're at a bit of a stalemate?

Reply:
Marcus is not part of Ubuntu. He can certainly weight in, but the main point persists, this is not an appropriate fix. Why not propose the correct solution?

Reply:
Both of you guys @dodys and @benhosmereop have valid points.

1 - This pull request fix an issue with how we handle packages from different distributions and their different names. You can think of this pull request targeting not only ubuntu, but every other distributions.
2 - This playbook is definitely not aligned with the bash implementation of Ubuntu's use case. For that, a new file called ubuntu.yml needs to be created and include ansible content that mirrors the bash remediation.

So, I suggest you @benhosmereop renaming the issue and pull request to something more generic and create a new issue that will describe the situation number 2 if you do not plan to work on this issue within this pull request.

Reply:
@ggbecker https://github.com/ComplianceAsCode/content/issues/11136 also

Reply:
@ggbecker still this fix will introduce a behavior not expected from some users, we either remove the multi_platform_all from it or we do one proper fix all at once. 
doing 2 steps won't help

Reply:
@dodys I'm trying to understand how you're arriving at the fact that this fix introduces something unexpected for users?

Prior to me updating it, the tasks and plays failed if the package wasn't named `NetworkManager`. If a user was using it prior to that, unless they specifically skipped those plays, the task failed anyway. If they were using it, and the package was called `network-manager` how is anything unexpected now going to happen?

The ansible tasks failed, which means it exited with an error, which was my original reason for creating this PR in the first place.

Reply:
@benhosmereop imagine any new user trying to get compliant to CIS or STIG. Imagine that they are server users that don't need network-manager and all of its dependencies, and they don't even have wireless on their server.
They today can:
1. run the bash and be good with it
2. run the ansible, which fails, but still they don't have network-manager installed and the audit of their system will pass, because they don't have wireless enabled.

If we merge your fix, what will happen is that for those users it will end up installing network-manager and all of its dependencies just to run a command to disable wireless interfaces on a system that don't have wireless interfaces and by having network-manager together with some other networking manager you might end up disrupting their system, there will be conflicts that they will need to manually solve.

Therefore, your proposed fix is introducing a regression for others.

Reply:
Hi guys, unfortunately I couldn't follow this discussion on Friday, but I think the points are clarified.
Thank you all for the constructive discussion which certainly triggered some improvements opportunities in the project.

I saw the Ansible remediation in `shared.yml` is well aligned to the Bash remediation in `shared.sh` and, for this case, it is beneficial the flexibility this PR is bringing to the rule. I only have to confirm if using `{{{ platform_package_overrides.NetworkManager }}}` is the best approach considering this idea is relatively new in remediation.

The point mentioned by @dodys is because there is another bash remediation specific for `ubuntu` defined in `ubuntu.sh` file. In simple words, currently there is no approved Ansible remediation for Ubuntu. The Playbook was already failing before the PR and with this PR it is no longer failing but is installing "NetworkManager", which might be undesired for Ubuntu users.

So, considering the previous comments, I propose to update the `platform` header in `shared.yml` to:
 `# platform =  multi_platform_fedora,multi_platform_ol,multi_platform_rhel,multi_platform_sle`

- This way, these distros will get this flexibility while `ubuntu` won't use this Ansible remediation.
- This will unblock this PR to be merged since it won't install NetworkManager in `ubuntu` and will also fix an existing issue where the Playbook fails in Ubuntu.

A separate PR can be created to introduce an Ansible remediation called `ubuntu.yml`, which will be aligned to the Bash remediation in `ubuntu.sh`, in order to fix the https://github.com/ComplianceAsCode/content/issues/11136.